### PR TITLE
ci: install aad-pod-identity in kube-system namespace

### PR DESCRIPTION
**Pull request by chewong**
**Reason for Change**:

install aad-pod-identity in kube-system namespace during the test suite so that it works in an AKS cluster with limited egress traffic.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: test/e2e/node_test.go [hunk 1]
```diff
@@ -274,7 +274,6 @@ var _ = Describe("When managing identities from the underlying nodes", func() {
 
 		By(fmt.Sprintf("Ensuring both keyvault-identity and cluster-identity are assigned to %s", node.Name))
 		userAssignedIdentities := azureClient.ListUserAssignedIdentities(node.Spec.ProviderID)
-		Expect(len(userAssignedIdentities) == 2).To(BeTrue())
```
Comment (chewong) [inline test/e2e/node_test.go:277]: Removing this line because clusters can have external user-assigned identities that are not part of the e2e test suite.

**Comment by chewong** [comment]
AKS soak test result: https://dev.azure.com/azure/aad-pod-identity/_build/results?buildId=13337&view=results

File: test/e2e/e2e_suite_test.go [hunk 2]
```diff
@@ -109,16 +109,15 @@ func dumpLogs() {
...
 			Labels: map[string]string{
 				"app.kubernetes.io/component": component,
 			},
 		})
 
 		for _, pod := range podList.Items {
 			By(fmt.Sprintf("Dumping logs for %s scheduled to %s", pod.Name, pod.Spec.NodeName))
-			_, err := exec.KubectlLogs(kubeconfigPath, pod.Name, corev1.NamespaceDefault)
-			Expect(err).To(BeNil())
+			_, _ = exec.KubectlLogs(kubeconfigPath, pod.Name, framework.NamespaceKubeSystem)
```
Comment (aramase) [inline test/e2e/e2e_suite_test.go:120]: Why are we removing the check here?

File: test/e2e/framework/helm/helm_helpers.go [hunk 3]
```diff
@@ -77,8 +79,9 @@ func Upgrade(input UpgradeInput) {
 	args := append([]string{
 		"upgrade",
 		chartName,
-		"charts/aad-pod-identity",
+		"manifest_staging/charts/aad-pod-identity",
```
Comment (aramase) [inline test/e2e/framework/helm/helm_helpers.go:82]: +1

File: test/e2e/e2e_suite_test.go (hunk 2 above)
Comment (chewong) [reply test/e2e/e2e_suite_test.go:120]: I wanted to make sure the entire `AfterSuite` is run despite potential errors during `dumpLogs`, that's why I ignore the error here. But I realized I could wrap all the cleanup logic in a defer function. Let me include the change in a follow-up PR (I merged this PR before seeing this comment).

Comment (chewong) [reply test/e2e/e2e_suite_test.go:120]: I piggy ride the change in #734.

**Pull request merged**